codec_record.go

```diff
@@ -77,13 +91,28 @@ func decoderOfStruct(cfg *frozenConfig, schema Schema, typ reflect2.Type) ValDec
 			continue
 		}
 
+		if field.action == FieldSetDefault {
+			if field.hasDef {
+				fields = append(fields, &structFieldDecoder{
+					field:   sf.Field,
+					decoder: createDefaultDecoder(cfg, field.Type(), field.def, sf.Field[len(sf.Field)-1].Type(), defW, defR),
+				})
+			}
+
+			continue
+		}
```

Comment:
Here is how I'm using the new **default decoder** in the case of `decoderOfStruct`